### PR TITLE
snap: move to graphics-core22

### PR DESCRIPTION
**Pull request by Saviq**
Also bring `scripts/` in, rather than staging from the deprecated snap-launch.

### CODE REVIEW

File: scripts/bin/wayland-launch [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+{
+  until
+    inotifywait --event create "$(dirname "$1")"&
+    inotify_pid=$!
+    [ -O "$1" ]
+  do
+    wait "${inotify_pid}"
+  done
+  kill "${inotify_pid}"
+}
```
Comment (AlanGriffiths) [inline scripts/bin/wayland-launch:29]: This is the old, buggy, `wait_for()`

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -41,29 +35,28 @@ apps:
...
 
 plugs:
   wayland:
   network-bind:
-  graphics-core20:
+  graphics-core22:
     interface: content
     target: $SNAP/graphics
-    default-provider: mesa-core20
+    default-provider: mesa-core22/edge
```
[the remark this answers is not in the document]
Comment (Saviq) [reply snap/snapcraft.yaml:53]: Yeah, it won't build with that, though, as `base: core22` checks for its existence on the store. Until we publish to stable, that is.

**Review by AlanGriffiths — APPROVED** [review]
WFM

**Review by AlanGriffiths — COMMENTED** [review]
WFM

bors merge

**Review by AlanGriffiths — APPROVED** [review]

**Comment by AlanGriffiths** [comment]
bors retry

**Pull request merged**